Kernel/Devices/PCISerialDevice.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2021, Idan Horowitz <idan.horowitz@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Devices/PCISerialDevice.h>
+
+namespace Kernel {
+
+static SerialDevice* s_the = nullptr;
+
+void PCISerialDevice::detect()
+{
+    PCI::enumerate([&](const PCI::Address& address, PCI::ID id) {
+        if (address.is_null())
+            return;
+
+        // HACK: There's currently no way to break out of PCI::enumerate, so we just early return if we already initialized the pci serial device
+        if (is_available())
+            return;
+
+        for (auto& board_definition : board_definitions) {
+            if (board_definition.device_id != id)
+                continue;
+
+            auto bar_base = PCI::get_BAR(address, board_definition.pci_bar) & ~1;
+            // FIXME: We should support more than 1 PCI serial port (per card/multiple devices)
+            s_the = new SerialDevice(bar_base + board_definition.first_offset, 64);
```

Comment:
This pattern is so great, what happens if we have multiple PCI serial cards in the system?

Reply:
check the `HACK` comment a couple lines above, currently i just only initialize the first found serial card